marklogic/tests/test_marklogic.py

```diff
@@ -0,0 +1,207 @@
+
+    aggregator.assert_all_metrics_covered()
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures("dd_environment")
+def test_check_with_filters(aggregator):
+    # type: (AggregatorStub) -> None
+    check = MarklogicCheck('marklogic', {}, [INSTANCE_FILTERS])
+
+    check.check(INSTANCE_FILTERS)
+
+    for metric in HOST_STATUS_METRICS + SERVER_STATUS_METRICS + TRANSACTION_STATUS_METRICS:
+        aggregator.assert_metric(metric, count=1)
+    for metric in FOREST_STATUS_SUMMARY_METRICS + REQUESTS_STATUS_METRICS:
+        aggregator.assert_metric(metric, at_least=1)
+    for metric in STORAGE_HOST_METRICS:
+        aggregator.assert_metric(metric, count=2)
+        aggregator.assert_metric_has_tag(metric, 'forest_name:Security', count=1)
+    for metric in STORAGE_FOREST_METRICS:
+        aggregator.assert_metric(metric, count=10)
+
+    # Resource filter only
+    for metric in RESOURCE_STORAGE_FOREST_METRICS:
+        aggregator.assert_metric(metric, tags=['forest_name:Security'], count=1)
+    for metric in [
+        'marklogic.requests.query-count',
+        'marklogic.requests.total-requests',
+        'marklogic.requests.update-count',
+    ]:
+        aggregator.assert_metric(metric, tags=['server_name:Admin', 'group_name:Default'], count=1)
+
+    aggregator.assert_all_metrics_covered()
+
+    # Service checks
+    aggregator.assert_service_check('marklogic.can_connect', MarklogicCheck.OK, count=1)
+    aggregator.assert_service_check('marklogic.database.health', MarklogicCheck.OK, count=10)
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+def test_metadata_integration(aggregator, datadog_agent):
+    # type: (AggregatorStub, Any) -> None
+    c = MarklogicCheck('marklogic', {}, [INSTANCE])
+    c.check_id = 'test:123'
+    c.check(INSTANCE)
+
+    parsed_version = version.parse(MARKLOGIC_VERSION)
+    version_metadata = {
+        'version.scheme': 'marklogic',
+        'version.major': str(parsed_version.major),
+        'version.minor': str(parsed_version.minor),
+        'version.patch': str(parsed_version.post),
+        'version.raw': MARKLOGIC_VERSION,
+    }
+
+    datadog_agent.assert_metadata('test:123', version_metadata)
+    datadog_agent.assert_metadata_count(len(version_metadata))
+
+    # Service checks
+    aggregator.assert_service_check('marklogic.can_connect', MarklogicCheck.OK, count=1)
+    aggregator.assert_service_check('marklogic.database.health', MarklogicCheck.OK, count=10)
+    aggregator.assert_service_check('marklogic.forest.health', MarklogicCheck.OK, count=10)
+
+
+@pytest.mark.e2e
+def test_e2e(dd_agent_check):
+    # type: (Any) -> None
+    aggregator = dd_agent_check(INSTANCE, rate=True)
+
+    for metric in GLOBAL_METRICS:
+        aggregator.assert_metric(metric, count=2)
+    for metric in STORAGE_HOST_METRICS:
+        aggregator.assert_metric(metric, count=2)
+    for metric in STORAGE_FOREST_METRICS:
+        aggregator.assert_metric(metric, count=20)
```

Comment:
Using `count=` seems not very maintainable. 

I would suggest asserting the same thing as for `test_check`.

Also, you can extract the content of `test_check` into a `_assert_metrics` and use it for both `test_check` and `test_e2e`.